sway/commands/input/tool_mode.c

```diff
@@ -0,0 +1,73 @@
+#include <strings.h>
+#include "sway/commands.h"
+#include "sway/config.h"
+
+static void set_tool_mode(struct input_config *ic,
+		enum wlr_tablet_tool_type type, enum sway_tablet_tool_mode mode) {
+	for (int i = 0; i < ic->tools->length; i++) {
+		struct input_config_tool *tool = ic->tools->items[i];
+		if (tool->type == type) {
+			tool->mode = mode;
+			return;
+		}
+	}
+
+	struct input_config_tool *tool = calloc(1, sizeof(*tool));
```

Comment:
style: `sizeof(struct input_config_tool)`

Reply:
Hmm, the kernel style says this in section 14 ([link](https://www.kernel.org/doc/Documentation/process/coding-style.rst)):

> The preferred form for passing a size of a struct is the following:
>
 >     p = kmalloc(sizeof(*p), ...);
>
> The alternative form where struct name is spelled out hurts readability and
> introduces an opportunity for a bug when the pointer variable type is changed
> but the corresponding sizeof that is passed to a memory allocator is not.

This makes sense to me. While I can change this code to use `sizeof(struct input_config_tool)` etc. instead, in that case I think the divergence from the kernel style should be documented as part of CONTRIBUTING.md alongside other divergences.

Reply:
I personally don't have any strong opinion one way or the other, but I do value consistency. Since we do use the `sizeof(struct input_config_tool)` style throughout the codebase, I'd rather continue with that. I can submit a PR tomorrow to add a note to CONTRIBUTING,md with this divergence.

Reply:
Submitted #5672